core/src/utils/native/haptic.ts

```diff
@@ -1,24 +1,90 @@
-// Main types for this API
+import type {
+  HapticsPlugin,
+  NotificationType as CapacitorNotificationType,
+  ImpactStyle as CapacitorImpactStyle,
+} from '@capacitor/haptics';
+
+import { getCapacitor } from './capacitor';
+
+export enum ImpactStyle {
+  /**
+   * A collision between large, heavy user interface elements
+   *
+   * @since 1.0.0
+   */
+  Heavy = 'HEAVY',
+  /**
+   * A collision between moderately sized user interface elements
+   *
+   * @since 1.0.0
+   */
+  Medium = 'MEDIUM',
+  /**
+   * A collision between small, light user interface elements
+   *
+   * @since 1.0.0
+   */
+  Light = 'LIGHT',
+}
+
 interface HapticImpactOptions {
-  style: 'light' | 'medium' | 'heavy';
+  style: CapacitorImpactStyle;
+}
+
+export enum NotificationType {
+  /**
+   * A notification feedback type indicating that a task has completed successfully
+   *
+   * @since 1.0.0
+   */
+  Success = 'SUCCESS',
+  /**
+   * A notification feedback type indicating that a task has produced a warning
+   *
+   * @since 1.0.0
+   */
+  Warning = 'WARNING',
+  /**
+   * A notification feedback type indicating that a task has failed
+   *
+   * @since 1.0.0
+   */
+  Error = 'ERROR',
 }
 
 interface HapticNotificationOptions {
-  style: 'success' | 'warning' | 'error';
+  type: CapacitorNotificationType;
+}
+
+interface TapticEngine {
+  gestureSelectionStart: () => void;
+  gestureSelectionChanged: () => void;
+  gestureSelectionEnd: () => void;
 }
 
 const HapticEngine = {
-  getEngine() {
-    const win = window as any;
-    return win.TapticEngine || (win.Capacitor?.isPluginAvailable('Haptics') && win.Capacitor.Plugins.Haptics);
+  getEngine(): HapticsPlugin | undefined {
+    const tapticEngine = (window as any).TapticEngine;
+    if (tapticEngine) {
+      // Cordova
+      return tapticEngine;
```

Comment:
can we cast this as `HapticsPlugin` too?

Reply:
When `getEngine` is used, it will be of type `HapticsPlugin` (or `undefined`), regardless if it is from the taptic engine or the Capacitor haptics plugin. Manually casting won't change the behavior since we have an explicit return type on the method.